force-app/main/default/classes/MaintenanceRequestHelper.cls

```diff
@@ -0,0 +1,57 @@
+public with sharing class MaintenanceRequestHelper {
+
+    public static void createMaintenanceRequestRecord(List<Case> caseList) {
+        try{
+            List<Case> newCases = new List<Case>();
+            Integer cycle = getMaintenanceCycle(caseList) == null ? 0 : getMaintenanceCycle(caseList);
+            for(Case c: caseList) {
+                Case newCase = new Case();
+                newCase.Vehicle__c = c.Vehicle__c;
+                newCase.Status = 'New';
+                newCase.Origin ='Web';
+                newCase.Type = 'Routine Maintenance';
+                newCase.Subject = 'Clone of ' + c.Subject ;
+                newCase.Date_Reported__c = Date.today();
+                newCase.OldId__c = c.Id;
+                newCase.Date_Due__c = Date.today().addDays(cycle);
+                newCases.add(newCase);
+            }
+            if(newCases.size() > 0) {
+                insert newCases;
+                createEquipmentMaintenanceItems(newCases);
+            }
+        }catch(Exception e) {
+                System.debug('Exception occured: ' + e.getMessage());
+        }
+    }
+    
+    public static void createEquipmentMaintenanceItems(List<Case> newCases) {
+        List<Equipment_Maintenance_Item__c> equipmentMain = new List<Equipment_Maintenance_Item__c>();
```

Comment:
in such cases better to name equipmentsToInsert and in general all collections should have plural name, not singular one, like equipment**s** not equipment